### ISSUE TITLE
WINUI3 Receive another app postmessage exception

### ISSUE THREAD

**Issue by destinykz** [microsoft/WindowsAppSDK#4274]
### Describe the bug

Marshal.PtrToStringAnsi 

### Steps to reproduce the bug

1、 a  app called sendmessage
var messageBytes = Encoding.UTF8.GetBytes(message);
var messagePtr = Marshal.AllocHGlobal(messageBytes.Length + 1);
Marshal.Copy(messageBytes, 0, messagePtr, messageBytes.Length);
var targetHandle = NativeMethods.FindWindow(null, "AppDisplayName".GetLocalized());
if (targetHandle != IntPtr.Zero)
{
    NativeMethods.PostMessage(targetHandle, NativeMethods.WM_MY_MESSAGE, IntPtr.Zero, messagePtr);
    Marshal.FreeHGlobal(messagePtr);
}
2、b app receive 
var msg = Marshal.PtrToStringAnsi(lParam);     ///exception here

System.AccessViolationException:“Attempted to read or write protected memory. This is often an indication that other memory is corrupt.”

### Screenshots

![微信图片_20240319194759](https://github.com/microsoft/WindowsAppSDK/assets/14058021/94fe29e5-8d10-4725-b9e8-20d3e73bdfd7)

### NuGet package version

None

### Packaging type

Unpackaged

### IDE

Visual Studio 2022

**Comment by castorix** [comment]
A pointer in a process has no meaning in another process
Use methods like _System.IO.MemoryMappedFiles_ or _WM_COPYDATA_

**Comment by destinykz** [comment]
> A pointer in a process has no meaning in another process Use methods like _System.IO.MemoryMappedFiles_ or _WM_COPYDATA_

i tried WM_COPYDATA , but i can not receive WM_COPYDATA message at all, 

![image](https://github.com/microsoft/WindowsAppSDK/assets/14058021/4e1a9ffe-ddd8-4e2e-99e4-45e15d550d6b)

**Comment by castorix** [comment]
> i tried WM_COPYDATA , but i can not receive WM_COPYDATA message at all,

Yes, weird, I just tested with an old app and I only receive it in the same instance
But Memory MappedFiles work (I tested with similar code as VB code I had posted in WinForms in [this thread](https://learn.microsoft.com/en-us/answers/questions/594919/how-do-i-share-specific-pieces-of-information-betw?page=2))
